content/en/blog/2021/announcing-etcd-3.5.md

```diff
@@ -0,0 +1,380 @@
+When 3.3 node receives the lease checkpoint request, it will panic, failing to
+handle the unknown apply request (see [etcd server apply
+code](https://github.com/etcd-io/etcd/blob/v3.3.25/etcdserver/apply.go#L119-L170)).
+However, one may not use such lease checkpointer feature and is **willing to
+risk incompatible changes in order to perform emergency rollbacks** (e.g., major
+defect in new etcd versions). To ensure such rollbacks are easy and reliable,
+we've added a **downgrade API that validates, enables, and cancels etcd version
+downgrades**. See [YoyinZyc@'s code change (from
+Google)](https://github.com/etcd-io/etcd/pull/11715).
+
+etcd cluster membership is applied with the same level of consistency as writes,
+by requiring quorum agreement. But previously, member list call was directly
+served from the server's local data, which may be stale. Now, **etcd serves
+member list with a linearizable guarantee -- if the server is disconnected from
+quorum, the member list call will fail**. See [jingyih@'s code change (from
+Google)](https://github.com/etcd-io/etcd/pull/11639).
+
+**gRPC gateway endpoint is now stable under `/v3/*`**. gRPC gateway generates
+HTTP API to make etcd gRPC-based HTTP/2 protocol accessible via HTTP/1, as such:
+
+> curl -X POST -L http://localhost:2379/v3/kv/put -d '{"key": "Zm9v", "value": "YmFy"}'
+
+**etcd client now uses latest gRPC v1.32.0 and migrated to upstream balancer
+implementation** -- see [ptabor@'s code change (from
+Google)](https://github.com/etcd-io/etcd/pull/12671), and **requires a new
+import path `"go.etcd.io/etcd/client/v3"`**.
+
+## Bug fixes
+
+etcd reliability and correctness are of utmost importance. That is why we
+backport all critical bug fixes to previous etcd releases. That said, these are
+the notable bugs we identified (and fixed) during etcd 3.5 development: (1)
+lease objects piling up caused memory leaks, and the fix was to clear expired
+lease queue in the old leader -- see [tangcong@'s fix (from
+Tencent)](https://github.com/etcd-io/etcd/pull/11731), (2) ongoing compact
+operation caused deadlock in mvcc storage layer -- [tangcong@'s fix (from
+Tencent)](https://github.com/etcd-io/etcd/pull/11817), (3) etcd server restart
+had redundant backend database open operations and as a result, reloading
+40-million keys took over 5-minute, and the fix reduced the restart time by half
+-- [tangcong@'s fix (from Tencent)](https://github.com/etcd-io/etcd/pull/11779),
+(4) if etcd crashed before completing defragment, the next defragment operation
+might have read the corrupted file, and the fix was to ignore and overwrite the
+existing file -- see [jpbetz@'s fix (from
+Google)](https://github.com/etcd-io/etcd/pull/11613), (5) client cancelling
+watch does not signal server creating leaky watchers, and the fix was to
+explicitly send a cancel request to the server -- see [jackkleeman@'s fix (from
+Apple)](https://github.com/etcd-io/etcd/pull/11613).
```

Comment:
https://github.com/etcd-io/etcd/pull/11613 -> https://github.com/etcd-io/etcd/pull/11850

Reply:
i have submitted pr #384 to fix the above two typo issues.